src/app/downloads/downloads.component.ts

```diff
@@ -0,0 +1,154 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Observable } from 'rxjs/Observable';
+import { CastleService } from '../core';
+import { EpisodeModel, EpisodeMetricsModel, PodcastMetricsModel, INTERVAL_DAILY, FilterModel } from '../ngrx/model';
+import { castlePodcastMetrics, castleEpisodeMetrics, castleFilter } from '../ngrx/actions/castle.action.creator';
+
+@Component({
+  selector: 'metrics-downloads',
+  template: `
+    <prx-spinner *ngIf="isLoading"></prx-spinner>
+    <p class="error" *ngIf="error">{{error}}</p>
+    <metrics-downloads-chart *ngIf="podcastMetrics && podcastMetrics.length > 0"></metrics-downloads-chart>
+  `
+})
+export class DownloadsComponent implements OnInit {
+  episodeStore: Observable<EpisodeModel[]>;
+  allEpisodes: EpisodeModel[];
+  podcastMetricsStore: Observable<PodcastMetricsModel[]>;
+  podcastMetrics: PodcastMetricsModel[];
+  episodeMetricsStore: Observable<EpisodeMetricsModel[]>;
+  episodeMetrics: EpisodeMetricsModel[];
+  filterStore: Observable<FilterModel>;
+  filter: FilterModel;
+  isLoading = true;
+  error: string;
+
+  constructor(private castle: CastleService, public store: Store<any>) {
+    this.filterStore = store.select('filter');
+  }
+
+  ngOnInit() {
+    // dispatch some default values for the dates and interval
+    const today = new Date();
+    const utcEndDate = new Date(Date.UTC(today.getFullYear(), today.getMonth(), today.getDate(), 23, 59, 59, 999));
+    const utcBeginDate = new Date(utcEndDate.valueOf() - (14 * 24 * 60 * 60 * 1000) + 1);// 14 days prior at 0:0:0
+    this.filter = {
+      beginDate: utcBeginDate,
+      endDate: utcEndDate,
+      interval: INTERVAL_DAILY
+    };
+    this.store.dispatch(castleFilter(this.filter));
+
+    this.filterStore.subscribe((state: FilterModel) => {
```

Comment:
The reason I named that `state` here is so that I remembered to use `this.filter` rather than the instance of state that it had when setting up the subscription. Seemed like I ran into some pitfalls with that.

Reply:
👍